### PR TITLE
[gym][snapshot] fix gym `use_system_scm` option and add `use_system_scm` option to snapshot

### LINKED ISSUE #17767: build_app parameter use_system_scm is ignored
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

I'm really happy that you've added `use_system_scm` property to the scan and gym commands in previous version!
But it seems like it's not working correctly since the flag is not added to the xcodebuild command. There should be `-scmProvider system` flag added there to force the xcode to use the system git configuration. It's missing in the log, which results in the build error. See the log below:

```
[17:56:31]: -----------------------
[17:56:31]: --- Step: build_app ---
[17:56:31]: -----------------------
[17:56:31]: Resolving Swift Package Manager dependencies...
[17:56:31]: $ xcodebuild -resolvePackageDependencies -workspace Example.xcworkspace -scheme Example -configuration Release
[17:56:32]: ▸ Command line invocation:
[17:56:32]: ▸     /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -resolvePackageDependencies -workspace "Example.xcworkspace" -scheme "Example" -configuration Release
[17:56:34]: ▸ Resolve Package Graph
[17:56:34]: ▸ Fetching git@github.com:private/repo.git
[17:56:36]: ▸ xcodebuild: error: Could not resolve package dependencies:
[17:56:36]: ▸   Authentication failed because the credentials were rejected
Command line invocation:
    /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -resolvePackageDependencies -workspace "Example.xcworkspace" -scheme "Example" -configuration Release

Resolve Package Graph

Fetching git@github.com:private/repo.git

xcodebuild: error: Could not resolve package dependencies:
  Authentication failed because the credentials were rejected
```

Running `xcodebuild -resolvePackageDependencies -workspace Example.xcworkspace -scheme Example -configuration Release -scmProvider system` forces the xcodebuild to use the system git configuration and resolves the packages correctly.

I've looked around and could not see any possible workarounds to resolve this issue. 

### Environment

<details>
  <pre>
### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.15.7                                     |
| Ruby                        | 2.7.0                                       |
| Bundler?                    | true                                        |
| Git                         | git version 2.24.3 (Apple Git-128)          |
| Installation Source         | ~/.rvm/gems/ruby-2.7.0/bin/fastlane         |
| Host                        | Mac OS X 10.15.7 (19H15)                    |
| Ruby Lib Dir                | ~/.rvm/rubies/ruby-2.7.0/lib                |
| OpenSSL Version             | OpenSSL 1.1.1h  22 Sep 2020                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 12.2                                        |

### System Locale

| Error                       |
| --------------------------- |
| No Locale with UTF8 found 🚫 |

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.170.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin                 | Version | Update-Status |
| ---------------------- | ------- | ------------- |
| fastlane-plugin-aws_s3 | 1.8.0   | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.4.0        |
| rake                      | 12.3.3       |
| CFPropertyList            | 3.0.3        |
| concurrent-ruby           | 1.1.7        |
| i18n                      | 1.8.5        |
| minitest                  | 5.14.2       |
| thread_safe               | 0.3.6        |
| tzinfo                    | 1.2.8        |
| activesupport             | 5.2.4.4      |
| public_suffix             | 4.0.6        |
| addressable               | 2.7.0        |
| httpclient                | 2.8.3        |
| json                      | 2.3.1        |
| algoliasearch             | 1.27.5       |
| rubyzip                   | 2.3.0        |
| apktools                  | 0.7.4        |
| atomos                    | 0.1.3        |
| aws-eventstream           | 1.1.0        |
| aws-partitions            | 1.404.0      |
| aws-sigv4                 | 1.2.2        |
| jmespath                  | 1.4.0        |
| aws-sdk-core              | 3.110.0      |
| aws-sdk-kms               | 1.39.0       |
| aws-sdk-s3                | 1.86.0       |
| babosa                    | 1.0.4        |
| bundler                   | 2.1.4        |
| claide                    | 1.0.3        |
| fuzzy_match               | 2.0.4        |
| nap                       | 1.1.0        |
| netrc                     | 0.11.0       |
| ffi                       | 1.13.1       |
| ethon                     | 0.12.0       |
| typhoeus                  | 1.4.0        |
| cocoapods-core            | 1.10.0       |
| cocoapods-deintegrate     | 1.0.4        |
| cocoapods-downloader      | 1.4.0        |
| cocoapods-plugins         | 1.0.0        |
| cocoapods-search          | 1.0.0        |
| cocoapods-trunk           | 1.5.0        |
| cocoapods-try             | 1.2.0        |
| colored2                  | 3.1.2        |
| escape                    | 0.0.4        |
| fourflusher               | 2.3.1        |
| gh_inspector              | 1.1.3        |
| molinillo                 | 0.6.6        |
| ruby-macho                | 1.4.0        |
| nanaimo                   | 0.3.0        |
| xcodeproj                 | 1.19.0       |
| cocoapods                 | 1.10.0       |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| declarative               | 0.0.20       |
| declarative-option        | 0.1.0        |
| digest-crc                | 0.6.2        |
| unf_ext                   | 0.0.7.7      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20190701 |
| dotenv                    | 2.7.6        |
| emoji_regex               | 3.2.1        |
| excon                     | 0.78.1       |
| multipart-post            | 2.0.0        |
| ruby2_keywords            | 0.0.2        |
| faraday                   | 1.1.0        |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.7        |
| faraday_middleware        | 1.0.0        |
| fastimage                 | 2.2.0        |
| jwt                       | 2.2.2        |
| memoist                   | 0.16.2       |
| multi_json                | 1.15.0       |
| os                        | 1.1.1        |
| signet                    | 0.14.0       |
| googleauth                | 0.14.0       |
| mini_mime                 | 1.0.2        |
| uber                      | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.2        |
| google-api-client         | 0.38.0       |
| google-cloud-env          | 1.4.0        |
| google-cloud-errors       | 1.0.1        |
| google-cloud-core         | 1.5.0        |
| google-cloud-storage      | 1.29.1       |
| mini_magick               | 4.11.0       |
| plist                     | 3.5.0        |
| security                  | 0.1.3        |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.8        |
| slack-notifier            | 2.3.2        |
| terminal-notifier         | 2.0.0        |
| unicode-display_width     | 1.7.0        |
| terminal-table            | 1.8.0        |
| tty-screen                | 0.8.1        |
| tty-cursor                | 0.7.1        |
| tty-spinner               | 0.9.3        |
| word_wrap                 | 1.0.0        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.3.0        |
| xcpretty-travis-formatter | 1.0.0        |
| mime-types-data           | 3.2020.1104  |
| mime-types                | 3.3.1        |
| fastlane-plugin-aws_s3    | 1.8.0        |
| xcode-install             | 2.6.8        |

*generated on:* **2020-12-09**
</pre>
</details>

### PR DESCRIPTION
**Pull request by rogerluan**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
Resolves #17767

### Description
This PR is a follow up of #17644, that introduced the use_system_scm parameter to `gym` and `scan`.
The issue was that that option was not being forwarded to (or read by, actually) `xcodebuild`'s ``-showBuildSettings` command.

Prior to this PR:

```
$ be fastlane run gym use_system_scm:true 
[✔] 🚀 
[18:13:37]: -----------------
[18:13:37]: --- Step: gym ---
[18:13:37]: -----------------
Select Scheme: 
1. redacted1
2. redacted2
3. redacted3
4. redacted4
5. redacted5
6. redacted6
7. redacted7
8. redacted8
9. redacted9
?  6
[18:13:39]: Resolving Swift Package Manager dependencies...
[18:13:39]: $ xcodebuild -resolvePackageDependencies -workspace ./redacted.xcworkspace -scheme redacted6
```

After this PR:

```
$ be fastlane run gym use_system_scm:true
[✔] 🚀 
[18:14:52]: -----------------
[18:14:52]: --- Step: gym ---
[18:14:52]: -----------------
Select Scheme: 
1. redacted1
2. redacted2
3. redacted3
4. redacted4
5. redacted5
6. redacted6
7. redacted7
8. redacted8
9. redacted9
?  6
[18:14:54]: Resolving Swift Package Manager dependencies...
[18:14:54]: $ xcodebuild -resolvePackageDependencies -workspace ./redacted.xcworkspace -scheme redacted6 -scmProvider system
```

This PR also adds `use_system_scm` option to snapshot, to fix specs.

### Testing Steps

To test this branch, modify your Gemfile as:

```ruby
gem "fastlane", :git => "https://github.com/fastlane/fastlane.git", :branch => "rogerluan-fix-gym-use-system-scm"
```

Then you can test it by running gym/build_app command using the `use_system_scm` option, such as:

`bundle exec fastlane run gym use_system_scm:true`

And verifying that it appends `-scmProvider system` to the end of the xcodebuild command.

### CODE REVIEW

**Review by ainame — DISMISSED** [review]
Looks good to me👍 Nice catch!

**Comment by felixscheinost** [comment]
When will this be merged?

**Comment by plu** [comment]
Looks great, looking forward to see this merged and released. What's holding this back?

This option solves CI working with private SwiftPM packages referenced via `git@...`

**Comment by plu** [comment]
I've just tried the branch, it partially looks good:

```
[15:55:25]: $ xcodebuild -resolvePackageDependencies -workspace eBay\ Kleinanzeigen.xcworkspace -scheme eBay\ Kleinanzeigen -derivedDataPath /Users/jenkins/Library/Caches/ios-client/DerivedData/Test -scmProvider system
```

But then later on it generates a wrong command:

```
[15:55:50]: $ set -o pipefail && env NSUnbufferedIO=YES xcodebuild -workspace eBay\ Kleinanzeigen.xcworkspace -scheme eBay\ Kleinanzeigen -derivedDataPath /Users/jenkins/Library/Caches/ios-client/DerivedData/Test -scmProvider system -scmProvider system -destination 'platform=iOS Simulator,id=8744F718-E0FB-4530-8BE5-A847CD631BD7' -resultBundlePath '/Users/jenkins/workspace/iOS_main_pipeline_PR-3155/test-reports/unit/eBay Kleinanzeigen.xcresult' build test | tee '/Users/jenkins/workspace/iOS_main_pipeline_PR-3155/test-reports/unit/Kleinanzeigen-eBay Kleinanzeigen.log' | xcpretty  --report html --output '/Users/jenkins/workspace/iOS_main_pipeline_PR-3155/test-reports/unit/report.html' --report junit --output '/Users/jenkins/workspace/iOS_main_pipeline_PR-3155/test-reports/unit/report.junit' --report junit --output '/var/folders/py/hqhn8n_n0bn3v7jmr9j4631m0000gl/T/junit_report20210112-71370-z590it' 
[15:55:50]: ▸ Loading...
[15:55:51]: ▸ xcodebuild: error: option '-scmProvider' may only be provided once
```

Adding `-scmProvider system` twice.

**Comment by ainame** [comment]
@rogerluan Any idea on that^?

**Comment by rogerluan** [comment]
@plu thanks for the feedback! Could you provide the command you used and the full output of the command (running `--verbose`)? 🙏

**Comment by plu** [comment]
Please check out #17957, it target the branch of this PR. I've added some failing tests there and a fix for it.

**Comment by plu** [comment]
@ainame could you have another look, please? The changes/fixes from the other PR have landed here now. I have integrated the changes on our CI, and I can confirm that "real world" usage of `gym` and `scan` are working there fine now, using `use_system_scm: true`.

File: fastlane.gemspec [hunk 1]
```diff
@@ -102,7 +102,7 @@ Gem::Specification.new do |spec|
 
   # Development only
   spec.add_development_dependency('rake')
-  spec.add_development_dependency('rspec', '~> 3.9.0')
+  spec.add_development_dependency('rspec', '~> 3.10.0')
```
Comment (ainame) [inline fastlane.gemspec:105]: Note: This is to use `include(...).once` matcher.

https://github.com/fastlane/fastlane/pull/17957#discussion_r557052834

File: gym/lib/gym/generators/build_command_generator.rb [hunk 2]
```diff
@@ -39,7 +39,9 @@ def options
         options << "-destination '#{config[:destination]}'" if config[:destination]
         options << "-archivePath #{archive_path.shellescape}" unless config[:skip_archive]
         options << "-resultBundlePath '#{result_bundle_path}'" if config[:result_bundle]
-        options << "-scmProvider system" if config[:use_system_scm]
+        if config[:use_system_scm] && !options.include?("-scmProvider system")
```
Comment (ainame) [inline gym/lib/gym/generators/build_command_generator.rb:42]: I think we should improve the entire flow to compose xcodebuild options to no have duplicated options but this is fine for now; e.g. Clarify what `project_path_array` brings and use Hash to manage options and convert them to real command argument eventually.

Thank you for taking care of this!

**Review by ainame — APPROVED** [review]
LGTM again🚀  @joshdholtz will double-check and merge when he can!

**Review by joshdholtz — APPROVED** [review]
This looks great! Thank you so much for making this fix ❤

**Pull request merged**